### PR TITLE
[IOTDB-4695] GROUP LEVEL query de-duplicates result columns unexpected

### PR DESCRIPTION
**Pull request by RYH61**
see : [https://issues.apache.org/jira/browse/IOTDB-4695](https://issues.apache.org/jira/browse/IOTDB-4695)

图片

### CODE REVIEW

**Review by ericpai — COMMENTED** [review]
Better to add an IT case

**Comment by RYH61** [comment]
> Better to add an IT case

ok~

File: integration-test/src/test/java/org/apache/iotdb/db/it/aggregation/IoTDBAggregationByLevelIT.java [hunk 1]
```diff
@@ -674,6 +675,25 @@ public void TestGroupByLevelWithoutAggregationFunc() {
...
+    String[] retArray = new String[] {"8", "8"};
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+
+      int cnt = 0;
+      try (ResultSet resultSet =
+          statement.executeQuery(
+              "select count(status),count(status) from root.*.* GROUP BY level=0")) {
+        while (resultSet.next()) {
+          String ans = resultSet.getString(count("root.*.*.status"));
```
Comment (ericpai) [inline integration-test/src/test/java/org/apache/iotdb/db/it/aggregation/IoTDBAggregationByLevelIT.java:689]: It seems that this case can't verify whether the output has 2 or 1 columns. You should use `resultSet.getMetadata()` and check
1. It contains 2 columns.
2. The 1st and 2nd column name are both `count(root.*.*.status)`.

**Review by ericpai — CHANGES_REQUESTED** [review]

File: integration-test/src/test/java/org/apache/iotdb/db/it/aggregation/IoTDBAggregationByLevelIT.java (hunk 1 above)
Comment (RYH61) [reply integration-test/src/test/java/org/apache/iotdb/db/it/aggregation/IoTDBAggregationByLevelIT.java:689]: `String[] retArray = new String[] {"8", "8"};`

The results that already contain two columns should be changed here

**Review by ericpai — APPROVED** [review]

File: integration-test/src/test/java/org/apache/iotdb/db/it/aggregation/IoTDBAggregationByLevelIT.java (hunk 1 above)
Comment (ericpai) [reply integration-test/src/test/java/org/apache/iotdb/db/it/aggregation/IoTDBAggregationByLevelIT.java]: Thanks. Maybe the while loop can be removed, and add another assert `assertEquals(count("root.*.*.status"), metadata.getColumn(1)); `

Comment (RYH61) [reply integration-test/src/test/java/org/apache/iotdb/db/it/aggregation/IoTDBAggregationByLevelIT.java]: ok~~

**Review by wangchao316 — APPROVED** [review]
LGTM

**Comment by ericpai** [comment]
All ClusterIT tasks are failed. Maybe there're some potential bugs.

**Review by ericpai — CHANGES_REQUESTED** [review]
All Cluster IT tasks are failed. Maybe there are some potential bugs.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java [hunk 2]
```diff
@@ -417,8 +424,9 @@ private List<Pair<Expression, String>> analyzeSelect(
           break;
         }
       }
+      outputExpressionMap.put(columnIndex, outputExpressions);
```
Comment (ericpai) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java:427]: I think the columnIndex is very useful in later use, so it's better to set from 0, not 1. If it begins at 0, we can easily get the corresponding ResultColumn by `queryStatement.getSelectComponent().getResultColumns().get(columnIndex)`

**Review by ericpai — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java (hunk 2 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java]: ok~~

**Pull request merged**